padl/transforms.py

```diff
@@ -1760,520 +1844,554 @@ def _add_name_to_splits(self, final_splits):
+
+        if 'preprocess' in build_splits:
+            self._preprocess_parents = _generate_parents_dict_from_edge_dict(self._preprocess_edges)
+            self._preprocess_list = _convert_to_structured_list(start_node=self.input_node,
+                                                                end_node=self.output_node,
+                                                                edges_dict=self._preprocess_edges,
+                                                                parents_dict=self._preprocess_parents,
+                                                                exclude_start_node=True,
+                                                                )
+            self._pd_preprocess = _build_transform_from_list(self._preprocess_list)
+
+        if 'forward' in build_splits:
+            self._forward_parents = _generate_parents_dict_from_edge_dict(self._forward_edges)
+            self._forward_list = _convert_to_structured_list(start_node=self.input_node,
+                                                             end_node=self.output_node,
+                                                             edges_dict=self._forward_edges,
+                                                             parents_dict=self._forward_parents,
+                                                             exclude_start_node=True,
+                                                             )
+            self._pd_forward = _build_transform_from_list(self._forward_list)
+
+        if 'postprocess' in build_splits:
+            self._postprocess_parents = _generate_parents_dict_from_edge_dict(self._postprocess_edges)
+            self._postprocess_list = _convert_to_structured_list(start_node=self.input_node,
+                                                                 end_node=self.output_node,
+                                                                 edges_dict=self._postprocess_edges,
+                                                                 parents_dict=self._postprocess_parents,
+                                                                 exclude_start_node=True,
+                                                                 )
+            self._pd_postprocess = _build_transform_from_list(self._postprocess_list)
+
+        return None, (self._pd_preprocess, self._pd_forward, self._pd_postprocess), True
 
-    def _pd_splits(self, input_components=0) -> Tuple[Union[int, List],
-                                                      Tuple[Transform, Transform, Transform],
-                                                      bool]:
-        """See the docstring of :meth:`Transform._pd_splits` for more details.
 
-        A parallel splits into:
-            - the parallel of its sub-transforms' preprocess
-            - the parallel of its sub-transforms' forward
-            - the parallel of its sub-transforms' postprocess
+class Compose(Pipeline):
```

Comment:
Have to say, pretty awesome how `Compose`, `Rollout` and `Parallel` became so much simpler with the new graph concept.